oga/core.py

```diff
@@ -61,28 +61,31 @@ def from_file(cls, file_path: Optional[str] = None) -> Optional["Config"]:
             root_dir=section.get("root_dir", fallback=default.root_dir))
 
 
-async def search(session: aiohttp.ClientSession, base_query: str) -> AsyncGenerator[str, None]:
+async def search(session: aiohttp.ClientSession, base_query: str, page_limit: Optional[int] = None) -> AsyncGenerator[str, None]:
     page = 0
 
     async def fetch() -> bytes:
         url = f"{base_query}&page={page}"
         async with session.get(url) as response:
             return await response.read()
 
-    # Special case for first page since we may not continue
-    data = await fetch()
-    last_page = parse_last_search_page(data)
-    asset_ids = parse_search_results(data)
-    for asset_id in asset_ids:
-        yield asset_id
-
-    while page < last_page:
-        page += 1
+    if page_limit is None or page_limit > 0:
+        # Special case for first page since we may not continue
```

Comment:
nit: In case someone passes a negative `page_limit` we should let them know, instead of silently returning zero results.

```
if page_limit is not None and page_limit < 0:
    throw new ValueError("page_limit cannot be negative")
```

Reply:
Nice that you noticed it!

Reply:
Note that in Python the sintax to throw the exception would be:
``` python
raise ValueError("message")
````

I notice, however, that you use assertions in the rest of the program, instead of throwing exceptions manually.

Because this is an error which would be of the programmer, and thus makes no sense calling for, I will use an assertion too.
If using an exception makes more sense tell me.

Reply:
Note, however, that the reason for the if existing wasn't to check that page_limit was positive, but simply that it wasn't 0.
Because a page_limit of 0 absolutely makes sense. It might be useless but its semantic is clear and there is an obvious expected result, which is an empty set. So I do need an empty set in that case.

So I added the assert (or exception throwing if it's better in this case), but the IF has to stand and still check if the value is not zero.

Reply:
Oops, that should have been `raise`.  Thanks!

> Note, however, that the reason for the if existing wasn't to check that page_limit was positive, but simply that it wasn't 0.

I'm in agreement, `page_limit=None` and `page_limit=0` are both acceptable values.  The extra assertion is just to guard the impossible values from a silent return.

```
assert (page_limit is None or page_limit >= 0, "page_limit must be None or non-negative")